python/graphscope/deploy/kubernetes/resource_builder.py

```diff
@@ -1142,7 +1142,17 @@ def build_liveness_probe(self):
             "ETCDCTL_API=3 etcdctl --endpoints=http://[127.0.0.1]:%s get foo"
             % str(self._listen_client_service_port),
         ]
-        return ExecProbeBuilder(liveness_cmd, timeout=15, failure_thresh=8)
+        return ExecProbeBuilder(liveness_cmd, timeout=15, period=10, failure_thresh=8)
+
+    def build_readiness_probe(self):
+        return HttpProbeBuilder(
```

Comment:
```
return TcpProbeBuilder(self._listen_peer_service_port, timeout=15, period=10, failure_thresh=8)
```

Reply:
`Tcp Probe` works.

Reply:
> `Tcp Probe` works.

we can fix this mistake by hard code the port as `2381`, the `health/metrics` port. 😂

not sure whether it's a good idea check only by tcp connectedness.

Reply:
Considering hard code `2381` may cause the port conflict when `hostNetwork=True`, I prefer using `TcpProbe` rather than hard code `2181`. 

@sighingnow Any suggestions?

Reply:
> 

We launch etcd without specifiying a separate `--listen-metrics-urls`. And the `/health` path is routed on the _client endpoint_, rather than the _peer endpoint_. Thus the HTTP probe on `self._listen_peer_service_port` and `/health` won't work.

Muli-instance etcd cluster launchings in the following way:

- step 1: listen on peer endpoint (2380)
- step 2: connect to peers
- step 3: listen on the client endpoint

To proceed step 2, we need to tell the k8s service that the underlying etcd pod is ready, by checking if the port of `peer endpoint` is ready. The client URL is not listened at that stage.

Checking `tcp connectedness` is enough as a indicator for the readiness of the etcd cluster, as using the client endpoint of etcd to launch vineyard, we will first verify that the etcd client URL is ready (see https://github.com/alibaba/GraphScope/blob/main/python/graphscope/deploy/kubernetes/resource_builder.py#L683).

Reply:
Awesome~